### ISSUE TITLE
daemon-reload + restart of service using JoinsNamespaceOf broken

### ISSUE THREAD

**Issue by SjonHortensius** [systemd/systemd#7790]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> 236

### Used distribution

> Archlinux

### Unexpected behaviour you saw

> A service that uses JoinsNamespaceOf will break after executing `systemctl daemon-reload` and restarting the service saying 

```
Failed to set up mount namespacing: No such file or directory
Failed at step NAMESPACE spawning /bin/sh: No such file or directory
```

### In case of bug report: Steps to reproduce the problem

> Create two services, `test-base.service`:
```
[Unit]
Description=test base

[Service]
PrivateTmp=true
ExecStart=/bin/sh -c 'sleep 999'

[Install]
WantedBy=multi-user.target
```
> `test-dep.service`
```
[Unit]
Description=test dep
JoinsNamespaceOf=test-base.service

[Service]
PrivateTmp=true
ExecStart=/bin/sh -c 'sleep 999'

[Install]
WantedBy=multi-user.target
```

> Now start both test-base and test-dep (in that order). Everything works fine, and test-dep can be restarted. Now execute `systemctl daemon-reload; systemctl restart test-dep`. The test-dep fails with the above error message about the namespace, presumable because the daemon-reload makes systemd 'forget' the namespace

**Comment by yuwata** [comment]
Confirmed. It seems that the deserialization process creates multiple identical ExecRuntime objects, not sharing single object. We may need to change serialize/deserialization operation for ExecRuntime object.

**Comment by yuwata** [comment]
Fix is waiting in #8044.

**Comment by SjonHortensius** [comment]
@yuwata  thanks for the update!
